tiger/src/lib.rs

```diff
@@ -33,6 +33,8 @@
 #![forbid(unsafe_code)]
 #![warn(missing_docs, rust_2018_idioms)]
 
+extern crate alloc;
```

Comment:
If TTH can not be implemented without allocations, then it should be feature-gated, i.e. you should introduce a new feature in Cargo.toml (e.g. `tree-hash`) and add `#[cfg(feature = "tree-hash")]` where appropriate.